Feature/ctap 0002

axv3012 opened the pull request.

isaric commented:
Hi , you have a problem right at the beginning - merge conflicts. Did you create your branch locally and the push to origin? If so, then it's possible that you didn't update your master branch before creating the new branch. You have to take into account that once I merged your previous PR to master branch, the state of that branch was updated. If you didn't update your local repo then this means you created the new branch from the previous (pre-PR-merge) state.
We will fix this together on monday as I will be in office. We will use the git stash to do this.

axv3012 committed:
CTAP-0002: Extending the model

axv3012 committed:
CTAP-0002: Updating update.html

isaric commented on src/main/java/com/crudtest/service/user/impl/DefaultUserService.java:
```diff
@@ -16,20 +19,17 @@
     private UserRepository userRepository;
 
     List<User> users = new ArrayList<User>();
```
We don't need this list

isaric commented on src/main/java/com/crudtest/service/user/impl/DefaultUserService.java:
```diff
@@ -16,20 +19,17 @@
+
 
     @Override
     public void saveUser(User user) {
```
This method should contain the logic for setting the Date attributes as required.

isaric commented on src/main/java/com/crudtest/model/User.java:
```diff
@@ -13,10 +20,17 @@
-        public User(){}
+
+        @Column(name = "DATE_CREATED", updatable = false)
+        private Date dateCreated = new Date();
```
Please move the initialization of date created either to the constructor or the the user service

isaric commented on src/main/java/com/crudtest/controller/UserController.java:
```diff
@@ -34,10 +37,6 @@ public ModelAndView registration(){
     @RequestMapping(value="/", method = RequestMethod.POST)
     public ModelAndView addUser(@Valid User user, BindingResult bindingResult){
```
Why are you using @Valid annotation when there is no annotation on your User POJO or a separate Validator class that handles it.

axv3012 committed:
CTAP-0002:
1. Removed the ArrayList from DefaultUserService
2. Set the Date to DefaultUserService in saveUser method
3. Initialization of date created and modified moved to service
4. Removed @Valid from Controller, because at this moment I am not validating no content

axv3012 committed:
CTAP-0002:
Small changes in the UserController

axv3012 committed:
CTAP-0002:
Removed unused imports from UserCotnroller

isaric commented on src/main/java/com/crudtest/controller/UserController.java:
```diff
@@ -32,12 +30,8 @@ public ModelAndView registration(){
-        if(userExists!=null){
-            bindingResult.rejectValue("lastName","error.user","This last name already exists");
-        }
         if (bindingResult.hasErrors()){
```
If you are not validating then you do not need the binding result as well

isaric commented on src/main/java/com/crudtest/controller/UserController.java:
```diff
@@ -55,4 +50,17 @@ public ModelAndView viewUsers(){
     }
 
+    @RequestMapping(value = "/update", method = RequestMethod.POST)
+    public ModelAndView update(@ModelAttribute User updateobject){
```
why is this a POST method if you are not doing anything with the DB?

isaric commented on src/main/java/com/crudtest/model/User.java:
```diff
@@ -1,49 +1,79 @@
+
+    }
+    public User(String firstName, String lastName, Date dateCreated, Date dateModified){
+        this.firstName = firstName;
```
Do you need this constructor?

axv3012 committed:
CTAP-0002:
1. Removed BindingResults from UserController
2. Removed POST from UserController update method because it was unnecessary
3. Removed the User parameterized constructor, it is not needed

isaric reviewed (approved):
Ok, we are done with this PR! Stay tuned for the next issue!
